nesta/production/luigihacks/elasticsearchplus.py

```diff
@@ -19,15 +23,144 @@
 COUNTRY_TAG="terms_of_countryTags"
 PUNCTUATION = re.compile(r'[a-zA-Z\d\s:]').sub('', string.printable)
 
+def sentence_chunks(text, chunksize=2000, delim='. '):
+    """Split a string into chunks, but breaking only on 
+    the specified delimiter.
+    
+    Args:
+        text (str): A string to split into chunks
+        chunksize (int): Minimum chunk size to yield
+        delim (str): Delimiter to split chunks
+    Yields:
+        chunk (str): The smallest possible chunks (minimum sise 
+                     :pyobject:`chunksize`) of text.
+    """
+    chunks = [] 
+    for _text in text.split(delim): 
+        chunks.append(_text) 
+        if sum(len(c) for c in chunks) > chunksize: 
+            yield delim.join(chunks) 
+            chunks = [] 
+    if len(chunks) > 0: 
+        yield delim.join(chunks) 
+
+
+class MLStripper(HTMLParser):
+    """Taken from https://stackoverflow.com/questions/753052/strip-html-from-strings-in-python. Tested in _sanitize_html."""
+    def __init__(self):
+        self.reset()
+        self.strict = False
+        self.convert_charrefs= True
+        self.fed = []
+        super().__init__()
+    def handle_data(self, d):
+        self.fed.append(d)
+    def get_data(self):
+        return ''.join(self.fed)
+
+def strip_tags(html):
+    """Taken from https://stackoverflow.com/questions/753052/strip-html-from-strings-in-python. Tested in _sanitize_html"""
+    s = MLStripper()
+    s.feed(html)
+    return s.get_data()
+
+@retry(wait_random_min=20, wait_random_max=30, 
+       stop_max_attempt_number=10)
+def translate(text, translator, chunksize=2000, rate_limit=0.4):
+    """Translate texts of any length via the Google Translate API.
+    
+    Args:
+        text (str): text to translate to English.
+        translator: A Translator instance.
+        chunksize (int): Ideal chunk size of text. Note, text is
+                         chunked in sentences defined by '. '
+        rate_limit (float): Time in seconds to sleep in order to be
+                            kind to Google.
+    Returns:
+        {text, langs} ({str, set}): Translated text and set of 
+                                    detected languages.
+    """
+    time.sleep(rate_limit)
+    chunks = list(sentence_chunks(text, chunksize=chunksize))
+    texts, langs = [], set()
+    for t in translator.translate(chunks):
```

Comment:
I know `dest='en'` is the default language but it would be worth adding it as an argument here for clarity

Reply:
Agreed